### PR TITLE
Support VLAN-tagged packets in testCksum

### PR DESCRIPTION
**Pull request by mbulatova**
- Add ParseDataCheckVLAN function
- ParseL4ForIPv4, ParseL4ForIPv6 work correctly with
VLAN-tagged packets

### CODE REVIEW

File: packet/checksum.go [hunk 1]
```diff
@@ -117,26 +121,26 @@ func CalculatePseudoHdrIPv6UDPCksum(hdr *IPv6Hdr, udp *UDPHdr) uint16 {
...
 		}
 	} else if ipv6 != nil {
 		tcp, udp, _ := p.ParseAllKnownL4ForIPv6()
 		if tcp != nil {
-			p.GetTCPForIPv6().Cksum = SwapBytesUint16(CalculatePseudoHdrIPv6TCPCksum(p.GetIPv6()))
+			p.GetTCPForIPv6().Cksum = SwapBytesUint16(CalculatePseudoHdrIPv6TCPCksum(ipv6))
 		} else if udp != nil {
-			p.GetUDPForIPv6().DgramCksum = SwapBytesUint16(CalculatePseudoHdrIPv6UDPCksum(p.GetIPv6(), p.GetUDPForIPv6()))
+			p.GetUDPForIPv6().DgramCksum = SwapBytesUint16(CalculatePseudoHdrIPv6UDPCksum(ipv6, udp))
 		}
```
Comment (dkolistratova) [inline packet/checksum.go]: If we use -hwol flag in testScenario 2  and send icmp ipv6 packet, 2nd part will try to calculate hardware cksum and will fail, maybe it makes sense to include software variant here as for ipv4?

File: test/framework/main/stability.json [hunk 2]
```diff
@@ -221,6 +221,90 @@
...
+        {
+            "name": "cksums_ipv4_tcp_vlan",
+            "test-time": 60000000000,
+            "test-type": "TestTypeScenario",
+            "test-apps": [
+                {
+                    "image-name": "nff-go-test-cksum",
+                    "app-type": "TestAppGo",
+                    "exec-cmd": [
+                        "./testCksum", "-testScenario=1", "-hwol", "-ipv4", "-tcp", "-vlan",  "-number=1000000", "-inport=0", "-outport=1"
```
Comment (dkolistratova) [inline test/framework/main/stability.json]: Do we need also test configuration with no hwol flag in 1st part (only in 2nd)?

Comment (mbulatova) [reply test/framework/main/stability.json]: We want checks HW vs SW Checksums in both part 1 and part 2.
I will add -hwol to 1st part in not-vlan tests

File: packet/checksum.go (hunk 1 above)
Comment (mbulatova) [reply packet/checksum.go]: Done

File: packet/vlan.go [hunk 3]
```diff
@@ -139,3 +139,30 @@ func (packet *Packet) RemoveVLANTag() bool {
...
+
+// ParseDataCheckVLAN parses L3, L4 and fills the field packet.Data.
+// returns 0 in case of success and -1 in case of
+// failure to parse L3 or L4. VLAN presence is checked.
+func (packet *Packet) ParseDataCheckVLAN() int {
+	var pktTCP *TCPHdr
+	var pktUDP *UDPHdr
+	var pktICMP *ICMPHdr
+
+	pktIPv4, pktIPv6, _ := packet.ParseAllKnownL3CheckVLAN()
```
Comment (ifilippov) [inline packet/vlan.go]: Why without ARP?

Comment (mbulatova) [reply packet/vlan.go]: ParseData() does not parse ARP too
ARP packets has no user Data

Comment (ifilippov) [reply packet/vlan.go]: Ok, get it.

**Comment by ifilippov** [comment]
Approve

**Pull request merged**